### PR TITLE
Fix Provider compat CI and sslapitest with FIPS provider

### PR DESCRIPTION
**Pull request by t8m**

### CODE REVIEW

**Comment by paulidale** [comment]
I prefer this approach I think.

**Review by paulidale — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Merged to all the active branches. Thank you for the reviews.

File: test/sslapitest.c [hunk 1]
```diff
@@ -4948,10 +4948,14 @@ static int test_key_exchange(int idx)
             break;
 #  ifndef OPENSSL_NO_ECX
         case 4:
+            if (is_fips)
+                return TEST_skip("X25519 might not be supported by fips provider.");
```
Comment (xnox) [inline test/sslapitest.c:4952]: well "X25519 should not be supported by fips provider" is more accurate, as any instances that did claim such support are unapproved back then through to present.

Comment (t8m) [reply test/sslapitest.c:4952]: Currently should not, maybe in future it could be. I think this is ok as is.

**Pull request closed without merging**